### ISSUE TITLE
Conda 4 does respect anaconda-client url

### ISSUE THREAD

**Issue by srossross** [conda/conda#2939]
```
conda --debug search --over -c org conda
Using Anaconda Cloud api site http://localhost:4990/api
DEBUG:binstar:Found login token: /Users/sean/Library/Application Support/binstar/http%3A%2F%2Flocalhost%3A4990%2Fapi.token
DEBUG:conda.fetch:channel_urls=OrderedDict([('https://conda.anaconda.org/org/osx-64/', ('org', 1)), ('https://conda.anaconda.org/org/noarch/', ('org', 1))])
```

Anaconda client is set up for `http://localhost:4990/api`.  conda no longer respects this url, it is fetching from `https://conda.anaconda.org/org/osx-64/`

**Comment by mcg1969** [comment]
Do you happen to have `channel_alias` set anywhere? The problem here is that somehow `channel_alias` is not being replaced with `bs.domain`, where `bs` is the output of `get_binstar()`. Perhaps as a workaround you can try explicitly setting `channel_alias`.

**Comment by kalefranz** [comment]
@HugoTian this is a test case we need to write with `responses`.

**Comment by mcg1969** [comment]
@srossross, @kalefranz: yes, I am almost certain here that `channel_alias` is the problem. The current logic decouples `channel_alias` from the Anaconda server domain returned by `get_binstar().domain`.

What we need to do is to change the logic so that if `channel_alias` is not set in the user or system config files, that we set some sort of flag to indicate that `binstar_channel_alias` is permitted to define it.

**Comment by kalefranz** [comment]
We should fix this in both 4.0.x and 4.1.x.

**Comment by mcg1969** [comment]
Are we sure it's broken in both? I can certainly check.

**Comment by mcg1969** [comment]
I think my PR #2564 broke this to fix issues #2556 and #1911

**Comment by kalefranz** [comment]
I'm not sure at all. But if it is, just saying this is something we should back port/fix.

**Comment by mcg1969** [comment]
@kalefranz, I'm looking at 4.0.x, and I see that `binstar_channel_alias` clobbers the value of `channel_alias` if `add_anaconda_token` is True. So with 4.0.x, it is not possible for them to be distinct. In other words, 4.0.x is assuming that _if_ `channel_alias` is set, that it points to the Anaconda Server instance. This means that @srossross's particular use case does not require a fix. I propose we leave it as is, then, and not try and backport a fix to #2556 and #1911.
